src/snowflake/snowpark/modin/plugin/_internal/utils.py

```
@@ -1914,6 +1908,69 @@ def count_rows(df: OrderedDataFrame) -> int:
     return row_count
 
 
+def get_object_metadata_row_count(object_name: str) -> Optional[int]:
+    """
+    Get the row count of a table or materialized view from Snowflake's metadata.
+    This function uses "SHOW OBJECTS" to avoid running a COUNT query on the table.
+
+    Args:
+        object_name: The name of the object, which can be fully qualified.
+
+    Returns:
+        The number of rows in the table or None, if no object was found or if the
+        object name is invalid.
+    """
+    session = pd.session
+
+    try:
+        parts = parse_table_name(object_name)
+    except Exception:
+        # If the object name is invalid, return None rather than propagate the exception
+        return None
+
+    db, schema, table = None, None, None
+    current_db = session.get_current_database()
+    current_schema = session.get_current_schema()
+
+    if len(parts) == 1:
+        table = parts[0]
+        schema = current_schema
+        db = current_db
+    elif len(parts) == 2:
+        schema, table = parts[0], parts[1]
+        db = current_db
+    elif len(parts) == 3:
+        db, schema, table = parts[0], parts[1], parts[2]
+    else:
+        return None  # pragma: no cover
+
+    # These returns should never be hit; they should be covered by the parse_table_name
+    # function. Leaving them in just to be safe.
+    if not db:
+        return None  # pragma: no cover
+    if not schema:
+        return None  # pragma: no cover
+
+    try:
+        query = f"SHOW OBJECTS LIKE '{table}' IN SCHEMA {db}.{schema} LIMIT 1"
+        res = session.sql(query).collect(
+            statement_params=get_default_snowpark_pandas_statement_params()
+        )
+    except Exception:
```

Should we make the exception class more specific so we fail loudly on unexpected errors?

---

Narrowed it to SnowflakeSQLException; which is the one I expect the most